File: packages/typescript/src/sdk/parity.test.mjs
import test from "node:test";
import assert from "node:assert/strict";
import fs from "node:fs";
import { buildValidationError } from "./validation.mjs";

test("validation fixture parity", () => {
  const raw = fs.readFileSync("../../contracts/fixtures/validator-fixtures.json", "utf8");
  const fixtures = JSON.parse(raw);
  for (const c of fixtures.cases) {
    const e = buildValidationError(c.ruleId, c.path, c.severity);
    assert.equal(e.code, c.expectedCode);
    assert.equal(e.ruleId, c.ruleId);
    assert.equal(e.path, c.path);
  }
});

File: packages/go/sh/whoa/sutradhar/sdk/v1/parity_test.go
package v1

import (
	"encoding/json"
	"path/filepath"
	"os"
	"testing"
)

type fixture struct {
	Cases []struct {
		RuleID       string `json:"ruleId"`
		Path         string `json:"path"`
		Severity     string `json:"severity"`
		ExpectedCode string `json:"expectedCode"`
	} `json:"cases"`
}

func TestValidationFixtureParity(t *testing.T) {
	data, err := os.ReadFile(filepath.Join("..", "..", "..", "..", "..", "..", "..", "contracts", "fixtures", "validator-fixtures.json"))
	if err != nil {
		t.Fatalf("read fixtures: %v", err)
	}
	var fx fixture
	if err := json.Unmarshal(data, &fx); err != nil {
		t.Fatalf("parse fixtures: %v", err)
	}
	for _, c := range fx.Cases {
		e := BuildValidationError(c.RuleID, c.Path, c.Severity)
		if e.Code != c.ExpectedCode {
			t.Fatalf("expected code %s, got %s", c.ExpectedCode, e.Code)
		}
		if e.RuleID != c.RuleID || e.Path != c.Path {
			t.Fatalf("fixture mismatch for rule %s", c.RuleID)
		}
	}
}
